doc/classes/OS.xml

```diff
@@ -562,6 +562,13 @@
 				[b]Note:[/b] This method is implemented on Android, iOS, Linux, macOS and Windows.
 			</description>
 		</method>
+		<method name="load_shell_environment" qualifiers="const">
+			<return type="void" />
+			<description>
+				Loads the default shell and copies environment variables set by the shell startup scripts to the app environment.
+				[b]Note:[/b] This method is implemented on macOS for non-sandboxed applications only.
+			</description>
+		</method>
```

Comment:
If it's macOS specific, is there really much need for exposing it? Sounds more like a Godot editor implementation detail than a generic API users would make use of.

Reply:
It might be useful for custom tools that are running external executables or configurable with environment variables, but still use GUI (and app bundle).

Reply:
What if we always call this method on startup (not just in the editor)? Is there a downside to always calling it?

Reply:
Then, if we don't want to expose the method maybe we can expose a project setting to determine whether to call the method at startup or not.

Reply:
I guess project setting OK as well, not sure what's better. We have plenty of `OS` methods that are useful only on a specific platform.

Reply:
Changed it to use `application/run/load_shell_environment` project setting instead of public method. Without this (in combination with #96146), or hard-coding paths (#74221), .NET version of Godot is still 100% unusable with homebrew installed dotnet SDK.